vortex-ipc/src/messages.rs

```diff
@@ -186,15 +195,147 @@ impl<'a> WriteFlatBuffer for IPCArray<'a> {
             .collect_vec();
         let children = Some(fbb.create_vector(&children));
 
-        fba::Array::create(
+        let stats = compute_and_build_stats(fbb, self.1, self.0.stats());
+
+        fb::Array::create(
             fbb,
-            &fba::ArrayArgs {
+            &fb::ArrayArgs {
                 version: Default::default(),
                 has_buffer: column_data.buffer().is_some(),
                 encoding,
                 metadata,
+                stats: Some(stats),
                 children,
             },
         )
     }
 }
+
+/// Computes all stats and uses the results to create an ArrayStats table for the flatbuffer message
+fn compute_and_build_stats<'a>(
+    fbb: &'_ mut FlatBufferBuilder<'a>,
+    array: &'_ ArrayData,
+    to_compute: &[Stat],
+) -> WIPOffset<fb::ArrayStats<'a>> {
+    let primitive_ptype = match array.dtype() {
+        DType::Primitive(ptype, _) => Some(ptype),
+        _ => None,
+    };
+
+    let mut frequencies: HashMap<_, _> = to_compute
```

Comment:
This map/collect pattern was introduced to avoid a bunch of if-contains/else, not sure if I like it better.

Reply:
It's weird...

You can just create a mut fb::ArrayStatsArgs and then for each stat, get value as scalar, then match the stat and cast if required.